### PR TITLE
Error: java.lang.ClassCastException: org.treblereel.dto.Import cannot be cast to org.treblereel.Import

### PR DESCRIPTION
**Pull request by treblereel**
I am proposing this fix which also does a little code simplification

### CODE REVIEW

**Comment by niloc132** [comment]
Consider a test case like this too, to verify that not just package is used, but also any type that they are nested in?
```
package foo.bar;

public class A {
  public static class NestedWithSameName {
    //...
  }
}
```
```
package foo.bar;

public class B {
  public static class NestedWithSameName {
    //...
  }
}
```

**Comment by treblereel** [comment]
@niloc132 i think it's another case but let me double check it

File: jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/DeserializerGenerator.java [hunk 1]
```diff
@@ -25,30 +27,32 @@
...
         if (!TypeRegistry.containsDeserializer(Type.stringifyTypeWithPackage(beanType))) {
             try {
-            	generateSubTypesDeserializers(beanType, packageName);
+                generateSubTypesDeserializers(beanType);
                 TypeRegistry.addInActiveGenDeserializer(beanType);
                 new AptDeserializerBuilder(packageName, beanType, filer).generate();
                 TypeRegistry.registerDeserializer(Type.stringifyTypeWithPackage(beanType), ClassName.bestGuess(deserializerName));
                 TypeRegistry.removeInActiveGenDeserializer(beanType);
+            } catch (FilerException e) {
+                //should be skipped
```
Comment (vegegoku) [inline jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/DeserializerGenerator.java:44]: Isnt this the same fix as in https://github.com/DominoKit/domino-jackson/pull/39

Comment (treblereel) [reply jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/DeserializerGenerator.java:44]: crap, yeap, you right, i need to remove it

**Comment by vegegoku** [comment]
> Consider a test case like this too, to verify that not just package is used, but also any type that they are nested in?
> 
> ```
> package foo.bar;
> 
> public class A {
>   public static class NestedWithSameName {
>     //...
>   }
> }
> ```
> 
> ```
> package foo.bar;
> 
> public class B {
>   public static class NestedWithSameName {
>     //...
>   }
> }
> ```

We do have an issue here, I have made the test case for this, and both classes generated to the same class with the same name.

File: jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/DeserializerGenerator.java (hunk 1 above)
Comment (vegegoku) [reply jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/DeserializerGenerator.java:44]: This is not resolved yet, right?!

File: jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/SerializerGenerator.java [hunk 2]
```diff
@@ -24,30 +26,31 @@
...
-    	
         if (!TypeRegistry.containsSerializer(Type.stringifyTypeWithPackage(beanType))) {
             try {
-            	generateSubTypeSerializers(packageName, beanType);
+            	generateSubTypeSerializers(beanType);
                 TypeRegistry.addInActiveGenSerializer(beanType);
                 new AptSerializerBuilder(packageName, beanType, ObjectMapperProcessor.filer).generate();
                 TypeRegistry.registerSerializer(Type.stringifyTypeWithPackage(beanType), ClassName.bestGuess(serializerName));
                 TypeRegistry.removeInActiveGenSerializer(beanType);
+            } catch (FilerException e) {
```
Comment (vegegoku) [inline jackson-apt-processor/src/main/java/org/dominokit/jacksonapt/processor/SerializerGenerator.java:41]: same as https://github.com/DominoKit/domino-jackson/pull/43#discussion_r507057187

**Pull request merged**